app/src/main/java/dev/ukanth/iconmgr/util/Util.java

```diff
@@ -451,6 +462,46 @@ public static boolean isPackageExisted(Context context, String targetPackage) {
         return true;
     }
 
+
+    public static String getAuthorName(Context context, String targetPackage) {
+        PackageManager pm = context.getPackageManager();
+        try {
+            Signature[] sigs = context.getPackageManager().getPackageInfo(targetPackage, PackageManager.GET_SIGNATURES).signatures;
+            for (Signature sig : sigs)
+            {
+                /*
+                 * Get the X.509 certificate.
+                 */
+                final byte[] rawCert = sig.toByteArray();
+                InputStream certStream = new ByteArrayInputStream(rawCert);
+
+                try {
+                    CertificateFactory certFactory = CertificateFactory.getInstance("X509");
+                    X509Certificate x509Cert = (X509Certificate) certFactory.generateCertificate(certStream);
+                    String issuerDN = x509Cert.getIssuerDN().getName();
+                    Log.i("DN", issuerDN);
+
+                    String[] nameComponents = issuerDN.split(",");
+                    for (String nameComponent : nameComponents) {
+                        String[] keyValue = nameComponent.trim().split("=");
+                        if (keyValue.length == 2 && keyValue[0].trim().equalsIgnoreCase("CN")) {
+                            authorName = keyValue[1].trim();
+                            return authorName;
+                        }
+                    }
+                }
+                catch (CertificateException e) {
+                     e.printStackTrace();
+                }
+            }
+        } catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+        }
+        return "No author name found";
```

Comment:
Rather than returning this, you should return empty / null and handle it accordingly